modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/base_controller.rb

```diff
@@ -44,6 +44,24 @@ def request_representative
         def shared_form_validation(form_number)
           target_veteran
           validate_json_schema(form_number.upcase)
+          @rep_id = validate_registration_number!(form_number)
+        end
+
+        def validate_registration_number!(form_number)
+          return if form_number != '2122' # Placeholder until API-33481
+
+          base = form_number == '2122' ? 'serviceOrganization' : 'representative'
+          rn = form_attributes.dig(base, 'registrationNumber')
+          poa_code = form_attributes.dig(base, 'poaCode')
+          rep = ::Veteran::Service::Representative.where('? = ANY(poa_codes) AND representative_id = ?',
+                                                         poa_code,
+                                                         rn).first
```

Comment:
I think `.first` will default to ordering by the `representative_id`. Our default has been to order by `created_at: :desc`.